Lib/tempfile.py

```diff
@@ -848,10 +848,17 @@ def write(self, s):
         return rv
 
     def writelines(self, iterable):
-        file = self._file
-        rv = file.writelines(iterable)
-        self._check(file)
-        return rv
+        if self._max_size and not self._rolled:
+            it = iter(iterable)
+
+            for line in it:
+                self.write(line)
+                if self._rolled:
+                    break
+
+            return self._file.writelines(it)
+        else:
+            return self._file.writelines(iterable)
```

Comment:
This might be a bit simpler:
```suggestion
        if not self._max_size or self._rolled:
            return self._file.writelines(iterable)

        it = iter(iterable)
        for line in it:
            self.write(line)
            if self._rolled:
                break

        return self._file.writelines(it)
```